src/cogent3/core/new_sequence.py

```diff
@@ -1579,25 +1611,57 @@ def parent_coordinates(self) -> tuple[str, int, int, int]:
         return self._seq.seqid, self._seq.parent_start, self._seq.parent_stop, strand
 
 
+@register_deserialiser(get_object_provenance(Sequence))
+def deserialise_sequence(data) -> Sequence:
+    return Sequence.from_rich_dict(data)
+
+
 class ProteinSequence(Sequence):
     """Holds the standard Protein sequence."""
 
     # constructed by PROTEIN moltype
-    ...
+
+    @classmethod
+    def from_rich_dict(cls, data):
```

Comment:
I wonder if you can get away with not defining the `from_rich_dict()` method on all the subclasses, since they inherit that. Registering the deserialiser for each subclass may be sufficient!

Reply:
can indeed get away with it!